app/lib/actions/middleware/load_setting_values.rb

```diff
@@ -0,0 +1,36 @@
+module Actions
+  module Middleware
+    class LoadSettingValues < ::Dynflow::Middleware
+      # ::Actions::Middleware::LoadSettingValues
+      #
+      # A middleware to ensure we load current setting values
+
+      def delay(*args)
+        reload_setting_values
+        pass(*args)
+      end
+
+      def plan(*args)
+        reload_setting_values
+        pass(*args)
+      end
+
+      def run(*args)
+        reload_setting_values
+        pass(*args)
+      end
+
+      def finalize(*args)
+        reload_setting_values
+        pass(*args)
+      end
+
+      private
+
+      def reload_setting_values
+        return unless Gem::Dependency.new('', ">= 2.5").match?('', ::SETTINGS[:version])
```

Comment:
I'd use `Gem::Version` instead.
```suggestion
        return unless Gem::Version.new(::SETTINGS[:version]) >= Gem::Version.new('2.5')
```

Reply:
Done